### ISSUE TITLE
Naxx40 issues list

### ISSUE THREAD

**Issue by bkmangx1** [ZhengPeiRu21/mod-individual-progression#70]
Hey, thank you for the module and the chance to experience old Naxxramas again. This is a list of issues we have found within this raid.

Trash mobs:
- Necro Stalker - this creature has WOTLK loot, drops level 70-80 items.
- Sewage Slime - these creatures are level 80.
- There are many creatures that are using both 10 & 25 man spells from WOTLK. For example [Patchwork Golem](https://www.wowhead.com/wotlk/npc=16017/patchwork-golem) is using [War Stomp 10man](https://www.wowhead.com/wotlk/spell=27758/war-stomp) and [War Stomp 25man](https://www.wowhead.com/wotlk/spell=56427/war-stomp) both spells are used in quick succession so players take ~10000 damage instantly. I see that same [10 man spell id](https://www.wowhead.com/classic/spell=27758/war-stomp) was used in vanilla where it did much less damage.

Boss:
- Grand Widow Faerlina - not casting Rain of Fire spell.
- Maexxna - [Web Wrap](https://www.wowhead.com/classic/spell=28622/web-wrap) ability doesn't apply a debuff, it should stun player and deal periodic damage until web is destroyed.
- Grobbulus - I believe [Slime Spray](https://www.wowhead.com/classic/spell=28157/slime-spray) ability is doing WOTLK damage, players got hit for 5-7k damage. Poison Cloud might be wrong too.
- Gluth - Zombie Chows can be ignored because they don't move unless engaged. They should move towards the boss after spawning.
- Instructor Razuvious - Deathknight Understudy have no abilities when mind controlled.
- Gothik the Harvester - Creatures on living side are not actively engaging players. There are no creatures being spawned/resurrected on undead side.
- Kel'Thuzad - minions in first phase have very little hp, we were using autobalance with 20 players and these mobs [Unstoppable Abomination](https://www.wowhead.com/classic/npc=16428/unstoppable-abomination) [Soul Weaver](https://www.wowhead.com/classic/npc=16429/soul-weaver) had ~2000 health. Encounter doesn't reset correctly after a wipe, all the creatures from phase 1 are missing and boss can be attacked directly.

Other:
- Green slime in construct quarter is behaving wrong, it seems to be reducing players max hp and killing them in 2 ticks. It should function like lava and similar environmental damage.
- [Eye Stalk](https://www.wowhead.com/classic/npc=16236/eye-stalk). These mobs are missing, they should be in the next room after defeating Heigan.
- Quest [Echoes of War](https://www.wowhead.com/wotlk/quest=9033/echoes-of-war) cannot be completed, killing these creatures in Naxx40 doesn't count for objective.
- When player is offline and group disbands, they don't get teleported outside of raid. Game client crashes on loading screen when they try to log in.

**Comment by ZhengPeiRu21** [comment]
Thank you for the helpful report. I will work to resolve these issues; note that some of these are using shared logic with WotLK Naxx and are AzerothCore upstream behavior bugs.

**Comment by bkmangx1** [comment]
Yes, I'm aware its not a simple task when many spells were changed directly to WOTLK versions. Hopefully you can find some solutions that don't conflict with AC. The biggest problem is with the trash mob spells, some quarters are ok, but others just have packs of mobs that kill players in 1 hit.

**Comment by ZhengPeiRu21** [comment]
I briefly investigate the gate not opening issue and didn't find an obvious solution; I will do more extensive debug soon and place a work-around if needed since this is a progression blocking issue.

**Comment by AlvinZhu** [comment]
> I briefly investigate the gate not opening issue and didn't find an obvious solution; I will do more extensive debug soon and place a work-around if needed since this is a progression blocking issue.

I have a temporary solution:
https://github.com/AlvinZhu/mod-individual-progression/commit/8f18f49d1142256dd9b9480ed894231a7e9879dd

But I don't know why does this bug occur. As I mentioned in issue https://github.com/ZhengPeiRu21/mod-individual-progression/issues/95
There should be a better solution. 

this bug already fixed:
- [x] Quest [Echoes of War](https://www.wowhead.com/wotlk/quest=9033/echoes-of-war) cannot be completed, killing these creatures in Naxx40 doesn't count for objective.

these bug:
- [ ] When player is offline and group disbands, they don't get teleported outside of raid. Game client crashes on loading screen when they try to log in.
- [ ] [Master Craftsman Omarion](https://www.wowhead.com/classic/npc=16365) - this NPC gets killed by monsters when players enter the raid.

I fixed these too, I'll submit PRs when I'm done testing.

**Comment by ZhengPeiRu21** [comment]
Thanks so much AlvinZhu for submitting those fixes! I have also just pushed a few more fixes myself. Necro Stalker loot has been corrected, the Sapphiron door issue should be resolved, Noth summons have been corrected, and creatures should no longer be performing duplicate abilities.

I have converted the original post to a checklist to track which issues still need to be resolved. Thank you very much bkmangx1 for your testing and the extensive report!

**Comment by Day36512** [comment]
What ended up being the dbc change that fixed the client crash?

**Comment by Day36512** [comment]
So I have a number of these fixes done, the major issue is it requires custom spells in spell.dbc. How would you like me to move forward if I wanted to submit these fixes?

**Comment by ZhengPeiRu21** [comment]
> So I have a number of these fixes done, the major issue is it requires custom spells in spell.dbc. How would you like me to move forward if I wanted to submit these fixes?

I would prefer to keep the module usable without client-side edits (and in cases where edits are necessary, like crafting, make it optional.) Aside from things like spell names and visuals, most spell parameters can be changed at cast time using CastCustomSpell. It's also possible to change the spell at load time using the OnLoadSpellCustomAttr hook in a GlobalScript.
However we change things, it's also important to not break WotLK Naxx, which uses many of the same spells, so CastCustomSpell is probably the best option in most cases here. Is it possible to implement your changes this way?

**Comment by Day36512** [comment]
I will take a look at my options and get back to you! Looks like I'll need to do a bit more reading.

**Comment by jellemeeus** [comment]
>  [  ] Sewage Slime, Deathcharger Steed - these creatures are level 80.

Sewage Slime, Grobbulus spawns these at the start of Abom wing
Deathcharger Steed, horse that spawns when dismounting charger in military quarter

They spawned as lvl60ish mobs for me

**Comment by Grimfeather** [comment]
2.5 years later.
Can someone update me on this?
Is this still an issue?
Has this been abandoned?

This is beyond my ability,

**Comment by sogladev** [comment]
The following can be completed as they've been fixed in acore and/or in this module:

- [x] Sewage Slime, Deathcharger Steed - these creatures are level 80.
- [x] Eye Stalk. These mobs are missing, they should be in the next room after defeating Heigan.
- [x] Maexxna - Web Wrap ability doesn't apply a debuff, it should stun player and deal periodic damage until web is destroyed.
- [x] Grobbulus - I believe Slime Spray ability is doing WOTLK damage, players got hit for 5-7k damage. Poison Cloud might be wrong too
- [x] Instructor Razuvious - Deathknight Understudy have no abilities when mind controlled.
- [x] Gothik the Harvester - Creatures on living side are not actively engaging players. There are no creatures being spawned/resurrected on undead side.

**Comment by Grimfeather** [comment]
Thanks! I'll leave this open then for now.

I can see why the unstoppable abomination and soul weaver have very little health.
their health modifier is still 1. while others have modifiers of 50+

I'll look at the Green Slime killing players in 2 ticks

**Comment by Grimfeather** [comment]
I believe I found the problem with the Disease Cloud the Sewage Slimes use.
the slime was using the level 80 version.

**Comment by Grimfeather** [comment]
closing this.

outdated.
almost everything has been solved.
what remains I can't fix.
